### PR TITLE
Implement Display and Debug traits

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I realised that there were a lot of missing `Debug`/`Display` traits implementations. Going through it.

cc @sdroege @EPashkin

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -63,6 +63,40 @@ impl Drop for RectangleList {
     }
 }
 
+impl fmt::Debug for RectangleList {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.ptr.is_null() {
+            write!(f, "RectangleList")
```
Comment (sdroege) [inline src/context.rs]: Is this even valid? Judging from other code here it's not so we don't need to handle that here.

File: src/context.rs [hunk 2]
```diff
@@ -63,6 +63,40 @@ impl Drop for RectangleList {
...
+                } else {
+                    format!("[{}]",
+                            slice::from_raw_parts(ptr, len as usize)
+                                .iter()
+                                .map(|s| format!("{:?}", s))
+                                .collect::<Vec<_>>()
+                                .join(", "))
+                };
+
+                write!(f, "RectangleList {{
```
Comment (sdroege) [inline src/context.rs]: Always use the `Formatter::debug_*()` functions for such things, in this case `debug_list()` or even simpler

```rust
f.debug_tuple("RectangleList")
    .field(&*self)  // &* gives you a &[Rectangle]
    .finish()
```

**Comment by sdroege** [comment]
I disagree with all these `Display` impls but keep them if you really want. Just like in glib/etc they don't fulfill the purpose of `Display`.

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks, 👍 after rebase to master

**Comment by GuillaumeGomez** [comment]
Updated!

**Comment by sdroege** [comment]
Go for it then

**Comment by GuillaumeGomez** [comment]
Yes sir!

**Pull request merged**